### PR TITLE
Task/Deposit e2e testing

### LINKED ISSUE #252: Deposit e2e testing

### PR DESCRIPTION
**Pull request by Ditmir-Vega**
closes #252

### CODE REVIEW

File: apps/trading-e2e/src/integration/deposits.feature
Comment (campbellssource) [inline apps/trading-e2e/src/integration/deposits.feature:28]: suggestion non-blocking: "User is prompted to approve more if the deposit amount is greater than the approved amount (and approve !=0)"

**Comment by campbellssource** [comment]
fyi: non-blocking: these tests all assume that the asset you are depositing is ERC20 (which require approvals) in the future there may be some assets that don't require an approval

File: apps/trading-e2e/src/integration/deposits.feature
Comment (campbellssource) [inline apps/trading-e2e/src/integration/deposits.feature:64]: suggestion non-blocking: "use the 'Use Maximum' button to populate amount input with the balance in the connected wallet

File: apps/trading-e2e/src/integration/deposits.feature
Comment (campbellssource) [inline apps/trading-e2e/src/integration/deposits.feature:75]: suggestion non-blocking: "user is warned if the the amount to deposit is greater than what is available in the connected wallet"

File: apps/trading-e2e/src/integration/deposits.feature
Comment (campbellssource) [inline apps/trading-e2e/src/integration/deposits.feature:96]: Question non-blocking: I think we talked about just testing that we see confirmation that a deposit is in progress rather than waiting for the deposit to be complete

File: apps/trading-e2e/src/integration/deposits.feature
Comment (campbellssource) [inline apps/trading-e2e/src/integration/deposits.feature:91]: FYI: another use case is depositing to the treasury e.g. address `0x0`

**Review by campbellssource — CHANGES_REQUESTED** [review]

File: apps/trading-e2e/src/integration/auction-orders.feature
Comment (jtsang586) [inline apps/trading-e2e/src/integration/auction-orders.feature:1]: For scenarios with auctions I think it makes more sense to group them with the other deal ticket tests. 
So I think this can be deleted.

File: apps/trading-e2e/src/integration/deposits.feature
Comment (Ditmir-Vega) [reply apps/trading-e2e/src/integration/deposits.feature:28]: @campbellssource is there another journey here where the user will be prompted to approve more if the deposit is higher than approval amount ?

i thought the approval step is done prior to any deposit attempted

File: apps/trading-e2e/src/integration/auction-orders.feature
Comment (Ditmir-Vega) [reply apps/trading-e2e/src/integration/auction-orders.feature:1]: depending on how many tests in total there are in deal ticket it might be better to split them out in seperate feature files .
When running tests in parralel  you don't want the features to be too long to run

File: apps/trading-e2e/src/integration/deposits.feature
Comment (Ditmir-Vega) [reply apps/trading-e2e/src/integration/deposits.feature:12]: does this mean that if the user has set their approval limit to 0 , then they will be prompted to increase it ?

File: apps/trading-e2e/src/integration/deposits.feature
Comment (Ditmir-Vega) [reply apps/trading-e2e/src/integration/deposits.feature:91]: what is the difference between depositing to the treasury and depositing to another wallet .
Is there a difference in deposit journey ?

File: apps/trading-e2e/src/integration/deposits.feature
Comment (campbellssource) [reply apps/trading-e2e/src/integration/deposits.feature:12]: The approved amount is 0 by default. so this is more like a "prompt the user to approve for the first time"

File: apps/trading-e2e/src/integration/deposits.feature
Comment (campbellssource) [reply apps/trading-e2e/src/integration/deposits.feature:29]: it is this is if the first time approved I set it only to a bit more than my first deposit. but this scenario is a second attempt to deposit where the amount will over use my initial approved amount

File: apps/trading-e2e/src/integration/deposits.feature
Comment (campbellssource) [reply apps/trading-e2e/src/integration/deposits.feature:92]: no difference really was just letting you know some background

File: apps/trading-e2e/src/integration/deposits.feature
Comment (Ditmir-Vega) [reply apps/trading-e2e/src/integration/deposits.feature:12]: I think this is a different test than the one you just mentioned. This test just makes sure that the user cannot deposit if they are on 0 limit .

We can write a test to approve on the first time but we need capsule to always load a checkpoint as most likely in this case the user would have already approved the first time

File: apps/trading-e2e/src/integration/deposits.feature
Comment (mattrussell36) [reply apps/trading-e2e/src/integration/deposits.feature:10]: It might be better to have a separate `web3.feature` to test that case (and other web3 connection related things)

File: apps/trading-e2e/src/integration/deposits.feature
Comment (mattrussell36) [inline apps/trading-e2e/src/integration/deposits.feature:13]: Do we need a line for `Given I am on the deposit page` at the start of each of these scenarios?

File: apps/trading-e2e/src/integration/deposits.feature
Comment (mattrussell36) [inline apps/trading-e2e/src/integration/deposits.feature:33]: Should you not specify what asset you are selecting here?

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: apps/trading-e2e/src/integration/deposits.feature
Comment (Ditmir-Vega) [reply apps/trading-e2e/src/integration/deposits.feature:33]: not yet .. we can fill in this blank when we start automating this scenario. we'll need to ensure the asset here is always available to select and we don't know if this would require extra work to set up etc .

**Pull request closed without merging**
